certora/specs/ExitLiquidity.spec

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+methods {
+    function extSloads(bytes32[]) external returns bytes32[] => NONDET DELETE;
+
+    function supplyShares(MorphoHarness.Id, address) external returns uint256 envfree;
+    function borrowShares(MorphoHarness.Id, address) external returns uint256 envfree;
+    function collateral(MorphoHarness.Id, address) external returns uint256 envfree;
+    function virtualTotalSupplyAssets(MorphoHarness.Id) external returns uint256 envfree;
+    function virtualTotalSupplyShares(MorphoHarness.Id) external returns uint256 envfree;
+    function virtualTotalBorrowAssets(MorphoHarness.Id) external returns uint256 envfree;
+    function virtualTotalBorrowShares(MorphoHarness.Id) external returns uint256 envfree;
+    function lastUpdate(MorphoHarness.Id) external returns uint256 envfree;
+
+    function libMulDivDown(uint256, uint256, uint256) external returns uint256 envfree;
+    function libMulDivUp(uint256, uint256, uint256) external returns uint256 envfree;
+    function libId(MorphoHarness.MarketParams) external returns MorphoHarness.Id envfree;
+}
+
+// Check that it's not possible to withdraw more assets than what the user has supplied.
```

Comment:
> let's say that there is a wrong direction rounding in the supply function

Yes, this is mostly checking the withdraw function rounding.

> and add a rule checking that the user cannot own more than what he supplied

Not sure how this rule would be helpful, because how can you define what you own if it's not what you can withdraw ?

> isn't it redundant anyway with the supplyWithdraw() rule

I don't think so because this works for more complex scenarios with potentially multiple prior interactions

Reply:
I'm changing the wording on this file

Reply:
>Not sure how this rule would be helpful, because how can you define what you own if it's not what you can withdraw ?

you define arbitrarily "what you own", and you verify that supply >= what you own >= withdraw